cloud/dimensiondata/dimensiondata_vlan.py

```diff
@@ -0,0 +1,411 @@
+                private_ipv4_range_size):
+    vlan_obj = get_vlan(module, driver, location, network_domain,
+                        vlan_id, name)
+    if type(vlan_obj) is dict:
+        module.exit_json(changed=False, msg="VLAN not found.")
+    else:
+        if vlan_obj.private_ipv4_range_size == private_ipv4_range_size:
+            return False
+        else:
+            vlan_obj.private_ipv4_range_size = private_ipv4_range_size
+            try:
+                return driver.ex_expand_vlan(vlan_obj)
+            except DimensionDataAPIException as e:
+                module.fail_json(msg="Failed to expand VLAN: %s" % e)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            region=dict(default='na', choices=dd_regions),
+            location=dict(required=True, type='str'),
+            network_domain=dict(required=True, type='str'),
+            name=dict(default=False, type='str'),
+            vlan_id=dict(default=False, type='str'),
+            description=dict(default='', type='str'),
+            private_ipv4_base_address=dict(default=False, type='str'),
+            private_ipv4_prefix_size=dict(default=False, type='str'),
+            action=dict(default='create', choices=['create', 'read', 'get',
+                                                   'update', 'delete',
+                                                   'expand']),
+            verify_ssl_cert=dict(required=False, default=True, type='bool'),
+            wait=dict(required=False, default=False, type='bool'),
+            wait_time=dict(required=False, default=600, type='int'),
+            wait_poll_interval=dict(required=False, default=2, type='int')
+        )
+    )
+
+    if not HAS_LIBCLOUD:
+        module.fail_json(msg='libcloud is required for this module.')
+
+    # set short vars for readability
+    credentials = get_credentials()
+    if credentials is False:
+        module.fail_json(msg="User credentials not found")
+    user_id = credentials['user_id']
+    key = credentials['key']
+    region = 'dd-%s' % module.params['region']
+    location = module.params['location']
+    network_domain_name = module.params['network_domain']
+    name = module.params['name']
+    vlan_id = module.params['vlan_id']
+    description = module.params['description']
+    base_address = module.params['private_ipv4_base_address']
+    prefix_size = module.params['private_ipv4_prefix_size']
+    verify_ssl_cert = module.params['verify_ssl_cert']
+    action = module.params['action']
+
+    # Instantiate driver
+    libcloud.security.VERIFY_SSL_CERT = verify_ssl_cert
+    DimensionData = get_driver(Provider.DIMENSIONDATA)
+    driver = DimensionData(user_id, key, region=region)
+
+    # Get Network Domain Object
+    network_domain = get_network_domain_by_name(driver, network_domain_name,
+                                                location)
+
+    # Process action
+    if action == 'create':
+        if name is False:
```

Comment:
If you change the defaults i nthe arg_spec you'll change these conditionals to "is None" instead of is False.